app/src/main/java/xterminators/spellingbee/cli/CLIController.java

```diff
@@ -363,14 +420,115 @@ private void ranks() {
      * @param filePath the path to save the puzzle to
      */
     private void save(String filePath) {
-        // TODO: Implement Save to given file
+        //Create an object of the Gson class
+        Gson saved = new Gson();
+
+        char[] nonRequiredLetters = puzzle.getSecondaryLetters();
+
+        char[] baseWord = new char[7];
+        baseWord[6] = puzzle.getPrimaryLetter();
+        for(int i = 0; i < baseWord.length - 1; i++){
+            baseWord[i] = nonRequiredLetters[i];
+        }
+
+        String filename = "";
+
+        //This will create a title for the Json file consisting
+        // of the non-required letters followed by the required letter.
+        for (char c : baseWord){
+            filename = filename + c;
+        }
```

Comment:
Why are you building a new name? This function should save to the file at filePath.